src/main/java/org/oldbox/clippy/frames/GenerateReportFrame.java

```diff
@@ -0,0 +1,264 @@
+package org.oldbox.clippy.frames;
+
+import com.intellij.uiDesigner.core.GridConstraints;
+import com.intellij.uiDesigner.core.GridLayoutManager;
+import com.intellij.uiDesigner.core.Spacer;
+import org.oldbox.clippy.Category;
+import org.oldbox.clippy.ClippyContext;
+import org.oldbox.clippy.NoteEntry;
+
+import javax.swing.*;
+import java.awt.*;
+import java.io.BufferedWriter;
+import java.io.File;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.nio.file.FileSystemException;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Set;
+
+public class GenerateReportFrame extends JFrame {
+    private JTextField destinationTextField;
+    private JButton chooseFileButton;
+    private JButton generateButton;
+    private JButton cancelButton;
+    private JPanel panel;
+    private JPanel selectDestinationPanel;
+    private JPanel actionPanel;
+    private JScrollPane listScrollPane;
+    private JList categoryList;
+    private JPanel categorySelectionPanel;
+    private JCheckBox selectAllCheckBox;
+
+    private int categoryCount;
+
+
+    public GenerateReportFrame() {
+        this.setTitle("Generate report");
+        this.setEnabled(true);
+        this.setVisible(true);
+        this.setDefaultCloseOperation(JFrame.DISPOSE_ON_CLOSE);
+
+
+        try {
+            Set<String> categories = ClippyContext.getInstance().getRepository().getCategories();
+
+            categoryCount = categories.size();
+            DefaultListModel model = new DefaultListModel();
+
+            categories.forEach(category -> model.addElement(category));
+            categoryList = new JList(model);
+            categoryList.setSelectionMode(ListSelectionModel.MULTIPLE_INTERVAL_SELECTION);
+            categoryList.setLayoutOrientation(JList.VERTICAL);
+            categoryList.setVisibleRowCount(-1);
+            listScrollPane.setViewportView(categoryList);
+            listScrollPane.setSize(getScrollPaneDimension(categories));
+
+
+        } catch (FileSystemException e) {
+            e.printStackTrace();
+        }
+
+        chooseFileButton.addActionListener(e -> chooseReportDestination());
+        cancelButton.addActionListener(e -> this.dispose());
+        generateButton.addActionListener(e -> generateReport());
+
+        this.add(panel);
+        this.pack();
+    }
+
+    private Dimension getScrollPaneDimension(Set<String> categories) {
+        int height = categories.size() * 10;
+        int width = 0;
+        for (String category : categories) {
```

Comment:
This could be written as:

```
int width = categories.stream().map(String::length).max(Integer::compare).get();
```

Reply:
Thanks for the tip! I will change to that.